File: apps/clinical-narrative/lib/core/models/transcription_models.dart
library;

class TranscriptionSegment {
  TranscriptionSegment({
    required this.startSeconds,
    required this.endSeconds,
    required this.text,
    this.confidence,
  });

  final double startSeconds;
  final double endSeconds;
  final String text;
  final double? confidence;

  factory TranscriptionSegment.fromJson(Map<String, dynamic> json) {
    return TranscriptionSegment(
      startSeconds: (json['startSeconds'] as num?)?.toDouble() ?? 0,
      endSeconds: (json['endSeconds'] as num?)?.toDouble() ?? 0,
      text: json['text']?.toString() ?? '',
      confidence: (json['confidence'] as num?)?.toDouble(),
    );
  }
}

class TranscriptionResponse {
  TranscriptionResponse({
    required this.requestId,
    required this.text,
    required this.processingTimeMs,
    required this.provider,
    required this.language,
    required this.segments,
    required this.warnings,
    this.confidence,
    required this.metadata,
  });

  final String requestId;
  final String text;
  final int processingTimeMs;
  final String provider;
  final String language;
  final List<TranscriptionSegment> segments;
  final List<String> warnings;
  final double? confidence;
  final Map<String, dynamic> metadata;

  factory TranscriptionResponse.fromJson(Map<String, dynamic> json) {
    return TranscriptionResponse(
      requestId: json['requestId']?.toString() ?? '',
      text: json['text']?.toString() ?? '',
      processingTimeMs: (json['processingTimeMs'] as num?)?.toInt() ?? 0,
      provider: json['provider']?.toString() ?? '',
      language: json['language']?.toString() ?? '',
      segments: (json['segments'] as List?)
              ?.map((item) => TranscriptionSegment.fromJson(
                    Map<String, dynamic>.from(item as Map),
                  ))
              .toList() ??
          const [],
      warnings: (json['warnings'] as List?)
              ?.map((item) => item.toString())
              .toList() ??
          const [],
      confidence: (json['confidence'] as num?)?.toDouble(),
      metadata: json['metadata'] is Map<String, dynamic>
          ? Map<String, dynamic>.from(json['metadata'] as Map)
          : const {},
    );
  }
}

File: services/survey-backend/app/domain/models/transcription_models.py
from __future__ import annotations

from typing import Any, Dict, List, Optional

from pydantic import BaseModel, Field, ConfigDict


class TranscriptionRequest(BaseModel):
    audio_base64: str = Field(..., alias="audioBase64")
    audio_format: str = Field(..., alias="audioFormat")
    sample_rate: Optional[int] = Field(default=None, alias="sampleRate")
    channels: Optional[int] = None
    duration_seconds: Optional[float] = Field(default=None, alias="durationSeconds")
    language: str = Field(default="pt-BR")
    clinical_mode: bool = Field(default=True, alias="clinicalMode")
    confidence_threshold: Optional[float] = Field(default=None, alias="confidenceThreshold")
    preview_text: Optional[str] = Field(default=None, alias="previewText")
    metadata: Dict[str, Any] = Field(default_factory=dict)

    model_config = ConfigDict(extra="forbid", populate_by_name=True)


class TranscriptionSegment(BaseModel):
    start_seconds: float = Field(..., alias="startSeconds")
    end_seconds: float = Field(..., alias="endSeconds")
    text: str
    confidence: Optional[float] = None

    model_config = ConfigDict(extra="forbid", populate_by_name=True)


class TranscriptionResponse(BaseModel):
    request_id: str = Field(..., alias="requestId")
    text: str
    confidence: Optional[float] = None
    segments: List[TranscriptionSegment] = Field(default_factory=list)
    processing_time_ms: int = Field(..., alias="processingTimeMs")
    provider: str
    language: str
    warnings: List[str] = Field(default_factory=list)
    metadata: Dict[str, Any] = Field(default_factory=dict)

    model_config = ConfigDict(extra="forbid", populate_by_name=True)
